src/thirtytwo/Window.cs

```diff
@@ -237,22 +237,28 @@ private LRESULT WindowProcedureInternal(HWND window, uint message, WPARAM wParam
                 break;
         }
 
-        bool handled = InvokeHandlers(out LRESULT result);
+        bool handled;
+        LRESULT result;
 
-        switch (message)
+        try
         {
-            case Interop.WM_NCDESTROY:
-                lock (_lock)
-                {
-                    // This should be the final message. Track that we've been destroyed so we know we don't have
-                    // to manually clean up.
+            handled = InvokeHandlers(out result);
+        }
+        catch
```

Comment:
This would be better handled as a `try` `finally`. Remove the case from the switch and just do the check for WM_NCDESTROY and handle it in the finally.